modules/swagger-parser-v3/src/main/java/io/swagger/parser/v3/OpenAPIV3Parser.java

```diff
@@ -39,17 +40,20 @@ public SwaggerParseResult readLocation(String url, List<AuthorizationValue> auth
 
             result = readWithInfo(url,auth);
 
-            String version = result.getOpenAPI().getOpenapi();
-            if(auth == null) {
-                auth = new ArrayList<>();
-            }
-            if(version != null && version.startsWith("3.0")) {
-                if (options != null) {
-                    if (options.isResolve()) {
-                        result.setOpenAPI(new OpenAPIResolver(result.getOpenAPI(), auth, null).resolve());
-                    }if(options.isResolveFully()){
-                        result.setOpenAPI(new OpenAPIResolver(result.getOpenAPI(), auth, null).resolve());
-                        new ResolverFully().resolveFully(result.getOpenAPI());
+            if (result.getOpenAPI() != null) {
```

Comment:
The statement `result.getOpenAPI()` is called many times, it would be better to make only one call, assigning to a variable and use this variable instead.